### ISSUE TITLE
Rendering issues with paths ...

### ISSUE THREAD

**Issue by Tom-Cuthill** [pauldendulk/Mapsui#159]
Unfortunately the platform for project I am working on is Windows 8.1 / RT, so I am not able to use the Skia library.

What I am seeing is that when the map resolution is under 20 metres, any geometry based on paths disappears.  You can see this behaviour in Mapsui.Samples.Windows8 if you pick demo three which displays a random polygon.  With an edge in view, if you continue to zoom in, the polygon suddenly disappears.  Using the inspector in Visual studio I can see that the hierarchy of canvases and their attached geometries and everything is as it should be (ie. the geometries of the polygon are still there, and if I apply the affine render transform manually, it resolves the coordinates from world coordinates into screen pixels correctly).  No exceptions are thrown.  I am at a loss to explain why the geometries don't render when the map resolution decreases.

The behaviour also happens with line strings.  Here is what I see at resolution = 38 metres:
![zoomout](https://cloud.githubusercontent.com/assets/24502866/24443574/c371ee4e-1495-11e7-9d4d-cd0cb04f438b.png)

And here is what I see when the resolution = 19 metres:

![zoomin](https://cloud.githubusercontent.com/assets/24502866/24443599/deb90c82-1495-11e7-995a-9241cb241516.png)

Note that this does not happen with all geometries, eclipses and rectangles behave correctly, it is just geometry collections with paths that exhibit this behaviour.

If anyone has any insights in how to get around this problem they would be most appreciated.  I am under a tight deadline, and I'm more or less blocked!!

Cheers,
Tom Cuthill

**Comment by charlenni** [comment]
I didn't looked into code, but make a suggestion.

Is there a ZoomMin and ZoomMax for the layer, so that it isn't displayed if you reach them?

**Comment by pauldendulk** [comment]
Do I understand it correctly you are using an older version of Mapsui that is using the Xaml renderer?

The Xaml renderer had problems like this. I assumed it was the consequence of some kind of performance optimization done by Microsoft. I was happy to drop it for that reason.  The problem is related to matrix scaling in WPF.

Or, do you see this problem in WPF as well?

**Comment by Tom-Cuthill** [comment]
Hi Paul,
Thanks for getting back to me.  Yes, I picked up the Mapsui.Rendering.Xaml - W8(Windows 8.1) project from a previous release so that we could use that platform.

I have tested both the UWP and WPF platforms and they work fine.  We'll have to decide if our clients are willing to shift platforms at this late stage.

I also wanted to offer a big thanks for your work on Mapsui.  The code is very well written, comprehensive and delight to work with :-)

Cheers,
Tom
